### DISCUSSION TITLE
Rate object enforces minute unit of time

### DISCUSSION THREAD

**Discussion by zanebclark** [openmobilityfoundation/curb-data-specification#81] [Data Specification]
John Good @FordAVLLC-jgood brought up a great point in the 2/15 meeting: The [Rate](https://github.com/openmobilityfoundation/curb-data-specification/tree/main/curbs#rate) object supports a dynamic `rate_unit`, but `increment_minutes`, `start_minutes `, and `end_minutes` are all still tied to a single unit of time.

**Comment by zanebclark** [comment]
We've introduced a `rate_unit` that was initially intended to reflect the unit of time associated with the `rate`, but it could dictate the unit of time used across the entire `Rate`:

| Name                 | Type    | Required/Optional | Description                                                                                                                                                                                                                                                           |
|----------------------|---------|-------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `rate`               | Integer | Required          | The rate for this space in cents (or the smallest denomination of local currency) per `rate_unit`.                                                                                                                                                                    |
| `rate_unit`          | Enum    | Required          | The unit of time associated with the `rate`. One of "second", "minute", "hour", "day", "week", "month", "year".                                                                                                                                                       |
| `increment_duration` | Integer | Optional          | If specified, this is the smallest number of `rate_unit`s a user can pay for (e.g., if `increment_duration` is `15` and `rate_unit` is `minute`, a user can pay for 15, 30, 45, etc. minutes).                                                                        |
| `increment_currency` | Integer | Optional          | If specified, the rate for this space is rounded up to the nearest increment of this amount, specified in the same units as `rate`.                                                                                                                                   |
| `start_duration`     | Integer | Optional          | The number of `rate_unit`s the vehicle must have already been present in the Curb Zone before this rate starts applying (_inclusive_, see [Range Boundaries](/general-information.md#range-boundaries)). If not specified, this rate starts when the vehicle arrives. |
| `end_duration`       | Integer | Optional          | The number of `rate_unit`s after which the rate stops applying (_exclusive_, see [Time Range](/general-information.md#time-range)). If not specified, this rate ends when the vehicle departs.                                                                        |

**Reply by schnuerle** [reply]
I worked in most of your suggestions here, thank you! I did leave `increment_amount` the same name, but added the word 'currency' in its description since currency in a field name in MDS tends to mean specifying the local currency, not the amount.

https://github.com/openmobilityfoundation/curb-data-specification/blob/release-1.0.0/curbs/README.md#rate

**Comment by FordAVLLC-jgood** [comment]
Thanks for bringing this up, @zanebclark. I was thinking that it would be good to future-proof the Curb Data Specification by allowing the use of seconds as a valid rate_unit. This would better accommodate use cases in high-demand urban curb frontages, and airports, where a lot of deliveries and drop-offs will happen.

**Reply by FordAVLLC-jgood** [reply]
Alternatively, we could change the field characteristics of increment_duration to allow for fractions of a minute (e.g. 1/10 minute = 6 seconds).

**Reply by zanebclark** [reply]
@FordAVLLC-jgood, `seconds` is already a valid `rate_unit`. Do you mean to say it would be good to future-proof by defining the remaining fields in terms of the designated `rate_unit` (e.g. `start_duration` interpreted in terms of the `rate_unit`)?
